src/Altinn.Platform/Altinn.Platform.Storage/Storage/Controllers/DataController.cs

```diff
@@ -298,22 +299,34 @@ private DataElement CreateDataElement(HttpRequest request, string elementType, I
                 theStream = request.Body;
                 contentType = request.ContentType;
             }
+            
+            DataElement newData = CreateDataElementHelper(elementType, instance, instanceGuid, creationTime, contentType, contentFileName, fileSize);
 
+            return newData;
+        }
+
+        /// <summary>
+        /// Creates a data element (helper method)
+        /// </summary>
+        /// <returns></returns>
+        internal DataElement CreateDataElementHelper(string elementType, Instance instance, Guid instanceGuid, DateTime creationTime, string contentType, string contentFileName, long fileSize)
```

Comment:
it is unclear what this helper function actually does. 
Consider renaming. If the function does many things consider moving the functionality to a helper class and split it into several smaller functions with names that describe what the function does.

Reply:
Fixed, created a new class, DataElementHelper.cs, which include this method with better naming as well.